python/lsst/daf/butler/formatters/jsonFormatter.py

```diff
@@ -78,7 +78,50 @@ def _writeFile(self, inMemoryDataset, fileDescriptor):
         with open(fileDescriptor.location.path, "w") as fd:
             if hasattr(inMemoryDataset, "_asdict"):
                 inMemoryDataset = inMemoryDataset._asdict()
-            json.dump(inMemoryDataset, fd)
+            fd.write(self._toBytes(inMemoryDataset).decode())
```

Comment:
Do not decode the bytes. Assuming these really are bytes, they should be written to the file as is since you have no formal idea what encoding is being used. Open the file in `wb` mode instead.